### PR TITLE
MueLu fence review

### PR DESCRIPTION
**Pull request by MicheldeMessieres**
@csiefer2 @kddevin 

This PR is not for submission, just for discussion.
It contains some comments about fences which will be used to guide ongoing work to address CUDA_LAUNCH_BLOCKING=0 issues.

### CODE REVIEW

**Comment by MicheldeMessieres** [comment]
Note that this Kokkos PR is relevant and some MueLu tests will fail without it:
[https://github.com/kokkos/kokkos/pull/3244](url)
When these kokkos changes are approved I'll open a Trilinos PR to sync.

File: packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp [hunk 1]
```diff
@@ -543,6 +543,7 @@ void GenerateRepresentativeBasisNodes(const Basis & basis, const SCFieldContaine
 
  basis.getValues(LoValues, ReferenceNodeLocations , Intrepid2::OPERATOR_VALUE);
 
+ // FENCE REVIEW: Fails without fence - MueLu_UnitTestsIntrepid2Tpetra_MPI_4
```
Comment (csiefer2) [inline packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp:546]: Line 564 below is an explicit UVM use.  So, yeah, we need a fence.

File: packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp [hunk 2]
```diff
@@ -604,6 +605,7 @@ void IntrepidPCoarsenFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Generat
   FC LoValues_at_HiDofs("LoValues_at_HiDofs",numFieldsLo,numFieldsHi);
   lo_basis.getValues(LoValues_at_HiDofs, hi_DofCoords, Intrepid2::OPERATOR_VALUE);
 
+  // FENCE REVIEW: Passing without fence - but UVM read of LoValues_at_HiDofs below indicates we need this fence
   Kokkos::fence(); // for kernel in getValues
```
Comment (csiefer2) [inline packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp:609]: Concur

File: packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp [hunk 3]
```diff
@@ -662,6 +664,7 @@ void IntrepidPCoarsenFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Generat
   FC LoValues_at_HiDofs("LoValues_at_HiDofs",numFieldsLo,numFieldsHi);
   lo_basis.getValues(LoValues_at_HiDofs, hi_DofCoords, Intrepid2::OPERATOR_VALUE);
 
+  // FENCE REVIEW: Passing without fence - but UVM read of LoValues_at_HiDofs below indicates we needs this fence
```
Comment (csiefer2) [inline packages/muelu/src/Transfers/PCoarsen/MueLu_IntrepidPCoarsenFactory_def.hpp:667]: Concur.

**Pull request closed without merging**